graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/caching/CacheControlInstrumentation.java

```diff
@@ -223,11 +238,13 @@ void replace(@Nullable CacheControlScope scope) {
 
     public Optional<String> maybeAsString() {
       Integer maxAgeValue = maxAge == null ? 0 : maxAge;
-      if (maxAgeValue.equals(0)) {
+
+      if (!allowZeroMaxAge && maxAgeValue.equals(0)) {
```

Comment:
actually I think we should change the logic here -> we should detect between `maxAge` is `null` and `0` so I think we probably should have the condition as

```java
if (maxAge == null || (!allowZeroMaxAge && maxAge == 0)) {
  return Optional.empty();
}
```

Reply:
Sounds good, I have added the explicit handling of `null` 👍